File: src/main/java/com/pisphere/governance/GovernanceToken.java
package com.pisphere.governance;

public class GovernanceToken {
    private String tokenName;
    private double totalSupply;

    public GovernanceToken(String tokenName, double totalSupply) {
        this.tokenName = tokenName;
        this.totalSupply = totalSupply;
    }

    public void mintTokens(double amount) {
        totalSupply += amount;
        System.out.println("Minted " + amount + " tokens. Total supply: " + totalSupply);
    }

    public double getTotalSupply() {
        return totalSupply;
    }
}

File: src/main/python/governance/governance_token.py
class GovernanceToken:
    def __init__(self, token_name, total_supply):
        self.token_name = token_name
        self.total_supply = total_supply

    def mint_tokens(self, amount):
        self.total_supply += amount
        print(f"Minted {amount} tokens. Total supply: {self.total_supply}")

    def get_total_supply(self):
        return self.total_supply
